packages/use-reapit-data/src/utils.ts

```diff
@@ -0,0 +1,113 @@
+import { GetAction, UpdateAction } from '@reapit/utils-common'
+import { AxiosError } from 'axios'
+import qs from 'qs'
+
+export interface StringMap {
+  [key: string]: string
+}
+
+export interface ReapitErrorField {
+  field?: string
+  message?: string
+}
+
+export interface ReapitError {
+  statusCode?: number
+  errors?: ReapitErrorField[]
+  description?: string
+  dateTime?: string
+}
+
+export const getMergedHeaders = (accessToken?: string, headers?: StringMap) => (): StringMap | null => {
+  return accessToken
+    ? {
+        Authorization: `Bearer ${accessToken}`,
+        'api-version': 'latest',
+        'Content-Type': 'application/json',
+        ...headers,
+      }
+    : null
+}
+
+export const handleReapitError = (error: AxiosError<any>, defaultMessage?: string): string => {
+  const { message } = error
+  const reapitError: ReapitError = error?.response?.data
+  const { description, errors } = reapitError ?? {}
+  const messageString = description
+    ? description
+    : message
+    ? message
+    : defaultMessage
+    ? defaultMessage
+    : 'An unknown error has occurred, please refresh the page and try again.'
+  const fieldErrors = errors?.map(({ field, message }) => `"${field}: ${message}"`)
+  const fieldString = fieldErrors ? fieldErrors.join(', ') : ''
+
+  return `${messageString} ${fieldString}`
+}
+
+export const getUrl = (action: GetAction | UpdateAction, queryParams?: Object, uriParams?: Object) => (): string => {
```

Comment:
I have an idea on how we can make our uriParams and body types changed depending on the type of request. Will need a refactor though so probably not best to do in this PR. We'll have to talk about it and make a ticket

Reply:
Sure, happy to discuss if you want to suggest something. Main thing was just to have no breaking changes initially...